bsoid_app/export_training.py

```diff
@@ -23,56 +17,51 @@ def __init__(self, WORKING_DIR, PREFIX, sampled_features, assignments, assign_pr
             self.assign_prob = assign_prob
             self.soft_assignments = soft_assignments
         except FileNotFoundError:
-            st.error('Cannot find training, please complete all the necessary steps')
+            print('Error: Cannot find training, please complete all the necessary steps')
 
     def save_csv(self):
-        if st.button('Generate training features and probabilities'):
             try:
-                if any('List of pose-relationship features' in o for o in self.options):
```

Comment:
You don't need this check?

Reply:
don't think I need this anymore because 'List of pose-relationship features' will definitely be included.    In the app, a user can select options of 'List of pose-relationship features', 'Mapping between features and assignment', 'Assignment probabilities'. Here I am just going to automatically generate all three. We could potentially make this also an option for users to provide a list of what they want, but there are already so many options it seemed unnecessary.